### ISSUE TITLE
Custom Feature Filtering 

### ISSUE THREAD

**Issue by Abbe98** [openstreetmap/iD#2580]
The user should be able to filter based on tag and key with the operators */!=/=. Multiply filters should be supported at the same time.

Currently custom iD instances has to modify `renderer/features.js` to get feature filtering working. 
An option to disable the default feature filtering(/or an API) together with custom filtering powered by presents/taginfo(which can be custom), would make the setup for custom instances shorter.

Example use case:
 - Multiply elements with `start_date` is currently visible on the map, we are only interested in elements tagged with `start_date=1959`, we could filter it with "start_date != 1959".

 - Or if someone is doing IOF mapping and need to filter away a ISSOM/ISOM line based on standard value("line=101").

Ugly mockup:

![ugly mockup](http://i1190.photobucket.com/albums/z458/abbe98/Namnloumls_zpsarp1esow.png)

**Comment by bhousel** [comment]
Cool idea.. I can see how you might need this for OHM. 

I kind of doubt we would put a custom querying language in here, but I will see if we can make the filters a bit more flexible like how presets/imagery/taginfo work.

See this too for a kind of hacky way to override the internal filters: https://github.com/openstreetmap/iD/issues/2539#issuecomment-76279840

**Comment by Abbe98** [comment]
We considered building a time-slider extension on top of current feature filtering but because iD does not have a plugin idea we moved over to the idea to contribute to core iD with this. Probably the best solution for implantation would be an API? The query support could be hidden by default.

**Comment by jameslaneconkling** [comment]
Is it possible to:
* control what filters are enabled on page load by default
* create custom filters

...when *redeploying* iD?  Or rather, any pointers on where to start poking around?  I'm guessing that the internals are pretty complicated.

(as background, I've [redeployed iD here]() to take advantage of custom presets, imagery, and min-zoom-level.  The project is focusing on editing of a narrow set of features--logging roads.  my idea is to disable loading of any feature except for `highway='track' && access='forestry'`, unless users specifically toggle those the checkboxes on again.)

**Comment by Abbe98** [comment]
Yes it's possible, the file you are looking for is [features.js](https://github.com/openstreetmap/iD/blob/master/js/id/renderer/features.js). It should be quite quite straight forward. Just create new boxes for `highway='track'`/`access='forestry'` and put any other tags under the "other" box. Then remove the "other" box from the UI... The last thing would be to trigger hiding of the "other" elements on load...

**Comment by jameslaneconkling** [comment]
@Abbe98 fantastic--thanks for the help.

**Comment by Abbe98** [comment]
@jameslaneconkling please let me know if you run into issues.

**Comment by jameslaneconkling** [comment]
Got it all hooked up.  The only hitch was triggering a click on the correct filter, as all HTML elements are built by d3 and hence not actually present at page load.  Found it worked if I placed the js in `js/id/ui.js`, after the function that builds the page elements.

In case anyone's interested--or has a less hackish approach--see the [changeset here](https://github.com/jameslaneconkling/iD/commit/be34b7e58203e6457d9e97d4011923ca9b3e5de0).
